fdbserver/IKeyValueStore.h

```diff
@@ -40,15 +40,27 @@ class IKeyValueStore : public IClosable {
 	virtual KeyValueStoreType getType() = 0;
 	virtual void set( KeyValueRef keyValue, const Arena* arena = NULL ) = 0;
 	virtual void clear( KeyRangeRef range, const Arena* arena = NULL ) = 0;
+	virtual Future<Void> commit(Version version, bool sequential = false) { return commit(sequential); }
```

Comment:
We should support set, clear, and commit at versions in order to future proof this for storage engines which can store all versions, not just exactly committed versions.  Rather than adding a version argument to each function, I think setWriteVersion() might be a better option, as it will change less often than the other methods are called.